third_party/fastpforlib/bitpackinghelpers.h

```diff
@@ -0,0 +1,977 @@
+  default:
+    break;
+  }
+}
+
+/*assumes that integers fit in the prescribed number of bits*/
+inline void fastpackwithoutmask(const uint32_t *__restrict in,
+                                uint32_t *__restrict out,
+                                const uint32_t bit) {
+  // Could have used function pointers instead of switch.
+  // Switch calls do offer the compiler more opportunities for optimization in
+  // theory. In this case, it makes no difference with a good compiler.
+  switch (bit) {
+  case 0:
+    internal::__fastpackwithoutmask0(in, out);
+    break;
+  case 1:
+    internal::__fastpackwithoutmask1(in, out);
+    break;
+  case 2:
+    internal::__fastpackwithoutmask2(in, out);
+    break;
+  case 3:
+    internal::__fastpackwithoutmask3(in, out);
+    break;
+  case 4:
+    internal::__fastpackwithoutmask4(in, out);
+    break;
+  case 5:
+    internal::__fastpackwithoutmask5(in, out);
+    break;
+  case 6:
+    internal::__fastpackwithoutmask6(in, out);
+    break;
+  case 7:
+    internal::__fastpackwithoutmask7(in, out);
+    break;
+  case 8:
+    internal::__fastpackwithoutmask8(in, out);
+    break;
+  case 9:
+    internal::__fastpackwithoutmask9(in, out);
+    break;
+  case 10:
+    internal::__fastpackwithoutmask10(in, out);
+    break;
+  case 11:
+    internal::__fastpackwithoutmask11(in, out);
+    break;
+  case 12:
+    internal::__fastpackwithoutmask12(in, out);
+    break;
+  case 13:
+    internal::__fastpackwithoutmask13(in, out);
+    break;
+  case 14:
+    internal::__fastpackwithoutmask14(in, out);
+    break;
+  case 15:
+    internal::__fastpackwithoutmask15(in, out);
+    break;
+  case 16:
+    internal::__fastpackwithoutmask16(in, out);
+    break;
+  case 17:
+    internal::__fastpackwithoutmask17(in, out);
+    break;
+  case 18:
+    internal::__fastpackwithoutmask18(in, out);
+    break;
+  case 19:
+    internal::__fastpackwithoutmask19(in, out);
+    break;
+  case 20:
+    internal::__fastpackwithoutmask20(in, out);
+    break;
+  case 21:
+    internal::__fastpackwithoutmask21(in, out);
+    break;
+  case 22:
+    internal::__fastpackwithoutmask22(in, out);
+    break;
+  case 23:
+    internal::__fastpackwithoutmask23(in, out);
+    break;
+  case 24:
+    internal::__fastpackwithoutmask24(in, out);
+    break;
+  case 25:
+    internal::__fastpackwithoutmask25(in, out);
+    break;
+  case 26:
+    internal::__fastpackwithoutmask26(in, out);
+    break;
+  case 27:
+    internal::__fastpackwithoutmask27(in, out);
+    break;
+  case 28:
+    internal::__fastpackwithoutmask28(in, out);
+    break;
+  case 29:
+    internal::__fastpackwithoutmask29(in, out);
+    break;
+  case 30:
+    internal::__fastpackwithoutmask30(in, out);
+    break;
+  case 31:
+    internal::__fastpackwithoutmask31(in, out);
+    break;
+  case 32:
+    internal::__fastpackwithoutmask32(in, out);
+    break;
+  default:
+    break;
+  }
+}
+
+inline void fastpackwithoutmask(const uint64_t *__restrict in,
```

Comment:
The `fastpackwithoutmask` is not used, right? Should we remove it, or do you think it will be useful for this and/or other compression methods in the future?

Reply:
It can be removed! It's for faster compression, but the speed difference is super minimal and I think it won't work for signed integers, I've removed the code!